### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by wisskirchenj**
- navigation drawer in Main Layout introduced
- category overview component realized with Vuetify VDataIterator
- new endpoint cardCount by category

### CODE REVIEW

File: frontend/src/feature/cards/composables/useCardsService.ts [hunk 1]
```diff
@@ -26,30 +29,44 @@ const useCardsService = () => {
     }
   }
 
-  const getCategories = async (errorResult: string = 'throw') => {
-    const url = 'http://127.0.0.1:8080/api/categories';
+  const getCards = async (categoryId: string, page: number, errorResult: string = 'throw') : Promise<Card[]> => {
+    const url = apiUrl + 'cards?categoryId=' + categoryId + '&page=' + page;
```
Comment (alfabravo2013) [inline frontend/src/feature/cards/composables/useCardsService.ts:31]: No string templates? I'm not against concatenation, just curious.

File: frontend/src/feature/cards/model/card.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export type Card = {
+  id: string,
```
Comment (alfabravo2013) [inline frontend/src/feature/cards/model/card.ts:2]: Card list is polymorphic and contains all flashcard types, which sets of fields are different. Each card item has `type` to tell what type it is.

File: frontend/src/feature/category/model/category.ts [hunk 3]
```diff
@@ -2,4 +2,6 @@ export type Category = {
   id: string,
   name: string,
   access: string,
+  numberOfCards?: number,
```
Comment (alfabravo2013) [inline frontend/src/feature/category/model/category.ts:5]: Should we add this information to the CategoryDTO?

**Review by alfabravo2013 — APPROVED** [review]

File: frontend/src/feature/cards/composables/useCardsService.ts (hunk 1 above)
Comment (wisskirchenj) [reply frontend/src/feature/cards/composables/useCardsService.ts:31]: yeah - funny I didn't use them here at the first place :-) Will change with next PR - it's nicer with template string

File: frontend/src/feature/category/model/category.ts (hunk 3 above)
Comment (wisskirchenj) [reply frontend/src/feature/category/model/category.ts:5]: we could of course - and I thought about it. I guess there are Pros and Cons. But I am open for both.

File: frontend/src/feature/cards/model/card.ts (hunk 2 above)
Comment (wisskirchenj) [reply frontend/src/feature/cards/model/card.ts:2]: yeah - and it is even easier in TS to model it, then in Java.
The Card type will change - I even left the comment in it - its WIP.

**Pull request merged**